diff -ru src/user/user.cpp src_patched/user/user.cpp
--- src/user/user.cpp	2020-06-22 17:51:39.000000000 +0200
+++ src_patched/user/user.cpp	2020-06-22 17:51:59.000000000 +0200
@@ -85,7 +85,7 @@
 
 void UserState::makeUserFile() {
   char buf[128];
-  sprintf(buf, "%s/%sXXXXXX", USERS_FILE_DIR, userName.c_str());
+  sprintf(buf, "%s/%sXXXXXX", USERS_FILE_DIR, userName.substr(0,100).c_str());
   /* Create users file if it does not exist */
   int userFileFD = mkstemp(buf);
   assert (userFileFD != -1);
